extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java

```diff
@@ -840,10 +846,71 @@ AdditionalBeanBuildItem authorizationController(LaunchModeBuildItem launchMode)
         return AdditionalBeanBuildItem.builder().addBeanClass(controllerClass).build();
     }
 
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    ObservedSecurityEventClassesBuildItem createObservedSecurityEventClassesSet(SecurityEventRecorder recorder) {
+        return new ObservedSecurityEventClassesBuildItem(recorder.createObservedEventClassesSet(),
+                recorder.createObservedEventClassesSet());
+    }
+
+    @Consume(RuntimeConfigSetupCompleteBuildItem.class) // producer requires runtime SecurityConfig
+    @Record(ExecutionTime.RUNTIME_INIT)
+    @BuildStep
+    SyntheticBeanBuildItem createDefaultSecurityEventProducer(SecurityEventRecorder recorder,
+            ObservedSecurityEventClassesBuildItem eventClassItem) {
+        return SyntheticBeanBuildItem
+                .configure(SecurityEventProducer.class)
+                .defaultBean()
+                .unremovable()
+                .setRuntimeInit()
+                .scope(Singleton.class)
+                .supplier(recorder.defaultSecurityEventProducer(eventClassItem.syncObserverEventClasses,
+                        eventClassItem.asyncObserverEventClasses))
+                .done();
+    }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    void findSecurityEventCdiObservers(SecurityEventRecorder recorder, SynthesisFinishedBuildItem synthesisFinished,
+            BeanArchiveIndexBuildItem indexBuildItem, ObservedSecurityEventClassesBuildItem eventClassItem) {
+        // collect all SecurityEvent classes
+        Set<DotName> knownSecurityEventClasses = new HashSet<>();
+        knownSecurityEventClasses.add(DotName.createSimple(SecurityEvent.class));
+        indexBuildItem
+                .getIndex()
+                .getAllKnownImplementors(SecurityEvent.class)
+                .stream()
+                .map(ClassInfo::name)
+                .forEach(knownSecurityEventClasses::add);
+
+        // keep only classes that have CDI observers
+        Set<String> syncObservedSecurityEventClasses = new HashSet<>();
+        Set<String> asyncObservedSecurityEventClasses = new HashSet<>();
+        for (DotName knownSecurityEventClass : knownSecurityEventClasses) {
+            for (ObserverInfo observer : synthesisFinished.getObservers()) {
+                boolean isObserved = observer.getObservedType().name().equals(knownSecurityEventClass);
```

Comment:
Just an FYI side note - you are matching based on type and events/observers can have qualifiers too. Technically, you can get a false positive here if users defined an observer with given type and some custom qualifier but it seems to be a very unlikely case here.